logbus/logstreamer/logstreamer.go

```diff
@@ -118,14 +119,27 @@ func (ls *LogStreamer) Write(delta *logstream.Delta) {
 	ls.mu.Lock()
 	ch := ls.ch // ls.ch may get replaced on retry
 	ls.mu.Unlock()
```

Comment:
should we also defer the Unlock here?

Reply:
No.. Channels are already concurrent in Go - they don't need a lock. And we don't want to hold the lock if the channel is congested as it could create a dead-lock.